src/Tmds.Ssh/ChannelContextSendMessageExtensions.cs

```diff
@@ -182,5 +182,55 @@ byte      SSH_MSG_CHANNEL_WINDOW_ADJUST
                 return packet.Move();
             }
         }
+
+        public static ValueTask SendChannelSubsystemMessageAsync(this ChannelContext context, string subsystem, CancellationToken ct)
+        {
+            return context.SendPacketAsync(CreatePacket(context, subsystem), ct);
+
+            static Packet CreatePacket(ChannelContext context, string subsystem)
+            {
+                /*
+                    byte      SSH_MSG_CHANNEL_REQUEST
+                    uint32    recipient channel
+                    string    "subsystem"
+                    boolean   want reply
+                    string    "name"
+                 */
+
+                using var packet = context.RentPacket();
+                var writer = packet.GetWriter();
+                writer.WriteMessageId(MessageId.SSH_MSG_CHANNEL_REQUEST);
+                writer.WriteUInt32(context.RemoteChannel);
+                writer.WriteString("subsystem");
+                writer.WriteBoolean(true);
+                writer.WriteString(subsystem);
+                return packet.Move();
+            }
+        }
+
+        public static ValueTask SftpInitMessageAsync(this ChannelContext context, uint version, CancellationToken ct)
+        {
+            return context.SendPacketAsync(CreatePacket(context, version), ct); // TODO later build sftp over SendChannelDataMessageAsync()
```

Comment:
And indeed, we should use `SendChannelDataMessageAsync`.

Reply:
For all SFTP or just these starting messages?

Reply:
using SendRequestAsync in the SftpClient directly isn't enough? :thinking:

Reply:
The init packet is a bit special, because it doesn't have a request id.